docs/madr/decisions/019-offline-signing-tokens.md

```diff
@@ -0,0 +1,230 @@
+# Offline signing tokens
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4031
+
+## Context and Problem Statement
+
+Kuma provides multiple tokens for different lines of security
+* User authenticating to Kuma API Server
+* Data plane proxy authenticating to Kuma DP Server
+* Zone proxy authenticating to Kuma DP Server
+
+Right now, a user has to use Kuma CP to issue the tokens.
+While this works, some users may prefer using their existing token issuer.
+We can provide an interface of a token (document required fields) and a way to provide public key of a token.
+This way a user can use different system to issue token, but let Kuma CP validate the token.
+This improves security posture if you already have a safe way of managing signing key for your tokens.
+
+## Considered Options
+
+* CP configuration and kumactl improvements
+
+## Decision Outcome
+
+Chosen option: "CP configuration and kumactl improvements".
+
+## Pros and Cons of the Options
+
+### User token
+
+#### CP configuration
+
+```yaml
+apiServer:
+  authn:
+    type: tokens
+    tokens:
+      bootstrapAdminToken: true
+      enableIssuer: true
+      validator:
+        useSecrets: true
+        publicKeys:
+        - kid: 123
+          keyFile: /tmp/key.pem # either keyFile or key can be defined
+          key: |
+            -----BEGIN RSA PUBLIC KEY-----
+            MIIEogIBAAKCAQEAy0KtfI7O0TJ00............
+            -----END RSA PUBLIC KEY-----
+          mapping:
+            username: user
+            groups: teams
+```
+
+*bootstrapAdminToken* - we already have it. Creates the admin token from the default signing key. Cannot be set to true if *enableIssuer* is set to false
+*enableIssuer* - if true you can generate tokens using Kuma CP API. It will also create a default signing key if one is missing.
+*validator.useSecrets* - if true, Kuma will try to access a Secret if the key is missing on a public keys list.
```

Comment:
Should we have a `secretName`? Or how it's going to locate the Secret?

Reply:
We have a hardcoded convention with a prefix. See
https://kuma.io/docs/2.0.x/security/dp-auth/#signing-key-rotation
https://kuma.io/docs/2.0.x/security/api-server-auth/#signing-key-rotation